lib/core/versions/latest/AnchorFile.ts

```diff
@@ -13,11 +14,18 @@ export default class AnchorFile {
    * Parses and validates the given anchor file buffer.
    * @throws `SidetreeError` if failed parsing or validation.
    */
-  public static parseAndValidate (anchorFileBuffer: Buffer, maxOperationsPerBatch: number): AnchorFileModel {
+  public static async parseAndValidate (anchorFileBuffer: Buffer, maxOperationsPerBatch: number): Promise<AnchorFileModel> {
+
+    let anchorFileDecompressedBuffer;
+    try {
+      anchorFileDecompressedBuffer = await Compressor.decompressBuffer(anchorFileBuffer);
+    } catch {
+      throw new SidetreeError(ErrorCode.AnchorFileDecompressionFailure);
+    }
 
     let anchorFile;
     try {
-      anchorFile = JSON.parse(anchorFileBuffer.toString());
+      anchorFile = JSON.parse(anchorFileDecompressedBuffer.toString());
```

Comment:
Consider:
```suggestion
      anchorFile = JSON.parse(anchorFileDecompressedBuffer);
```
Looks like you were able to do so for the batch file decompression.

Reply:
Good catch cause the BatchFile is using JsonAsync where as here we weren't. Updated the code to use JsonAsync now.